Test_Cases.py

```diff
@@ -1,169 +1,216 @@
+from selenium.webdriver.chrome.options import Options
+from selenium import webdriver
 import unittest
-from Yahoo_Page import *
-from Tools import *
-import Settings 
+from Yahoo_Page import YahooPage
+import Tools as tools
+import Settings as settings
+import time
 
 
 class HomePageSetup(unittest.TestCase):
 
     def setUp(self):
-    
-        self.driver = webdriver.Chrome(Settings.path_to_webdriver, options=options)  
-        self.yahoo_page = Yahoo_Page(self.driver)      
+        options = YahooPage.options
+        s = Service(settings.path_to_webdriver)
+        self.driver = webdriver.Chrome(service=s, options=options)
+        self.yahoo_page = YahooPage(self.driver)
 
     def tearDown(self):
         self.driver.quit()
 
 
+class ErrorMessagePasswords(HomePageSetup):
+
+    def random_password(self, password_len):
+        excluded_password_chars = ['\n', '\t', '\r', '\x0b', '\x0c']
+        return tools.generate_random_string(
+            password_len, excluded_password_chars)
+
+    def setUp(self):
+        super().setUp()
+
+        self.yahoo_page.login()
+        self.yahoo_page.navigate_to_security_tab()
+        self.yahoo_page.click_change_password_link()
+
+    def test_short_password_error_message(self):
+        """
+        Confirm that when the password is 8 characters that the data
+        error status is "WEAK_PASSWORD" Confirm the resulting error text
+        is "- Your password is too easy to guess."
+
+        acceptance criteria
+        --------------------
+        -When an 8 character password is entered, the data error status
+         and the error text exhibit's this is too short of a password.
+
+        Note
+        --------------------
+        -Yahoo security will randomly ask to "prove you are not a robot".
+         Which, at times, can be responsible for the tests in Error Message
+         Passwords class to error. There is not an easy fix for this.
+        """
+        yahoo_page = self.yahoo_page
+
+        yahoo_page.input_new_password_field(self.random_password(5))
+
+        self.assertEqual(yahoo_page.short_password_error_message().
+                         get_attribute('data-error'), "WEAK_PASSWORD")
+        self.assertEqual(yahoo_page.short_password_error_message().text,
+                         "- Your password is too easy to guess.")
+
+    def test_moderate_password_error_message(self):
+        """
+        Confirm that when the password 9 or 10 characters in length, the
+        error data status is "ALMOST_THERE". Confirm the resulting error
+        text is "- Almost there".
+
+        acceptance criteria
+        --------------------
+        -When a password that is between 9 and 10 characters is entered,
+         the data error status and the error text exhibit's this is a
+         moderate length for the password.
+        """
+        yahoo_page = self.yahoo_page
+
+        yahoo_page.input_new_password_field(self.random_password(9))
```

Comment:
The long comment below and another is my main worry with your thoughts on it's content and length.